### ISSUE TITLE
null value returned from native call not assignable to Object

### ISSUE THREAD

**Issue by karimiumar** [ceylon/ceylon-ide-intellij#504]
```
null value returned from native call not assignable to Object
ceylon.language.AssertionError "null value returned from native call not assignable to Object"
	at com.redhat.ceylon.compiler.java.Util.checkNull(Util.java:985)
	at com.redhat.ceylon.ide.common.model.asjava.getJavaQualifiedName_.getJavaQualifiedName(JObjectMirror.ceylon:96)
	at org.intellij.plugins.ceylon.ide.ceylonCode.model.scanJavaIndex_$1processor_.findOrCreateDeclaration$priv$(scanJavaIndex.ceylon:150)
	at org.intellij.plugins.ceylon.ide.ceylonCode.model.scanJavaIndex_$1processor_.process(scanJavaIndex.ceylon:362)
	at org.intellij.plugins.ceylon.ide.ceylonCode.model.scanJavaIndex_$1processor_.process(scanJavaIndex.ceylon)
	at com.intellij.psi.impl.search.AllClassesSearchExecutor.processClassesByNames(AllClassesSearchExecutor.java:84)
	at com.intellij.codeInsight.completion.AllClassesGetter.processJavaClasses(AllClassesGetter.java:162)
	at org.intellij.plugins.ceylon.ide.ceylonCode.model.scanJavaIndex_.scanJavaIndex(scanJavaIndex.ceylon:413)
	at org.intellij.plugins.ceylon.ide.ceylonCode.model.IdeaModule.getAvailableDeclarations(IdeaModule.ceylon:76)
	at com.redhat.ceylon.model.typechecker.model.Package.getMatchingDeclarations(Package.java:239)
	at com.redhat.ceylon.ide.common.util.SingleSourceUnitPackage.getMatchingDeclarations(SinceSourceUnitPackage.ceylon:107)
	at com.redhat.ceylon.model.typechecker.model.Element.getMatchingDeclarations(Element.java:244)
	at com.redhat.ceylon.model.typechecker.model.TypeDeclaration.getMatchingDeclarations(TypeDeclaration.java:894)
	at com.redhat.ceylon.model.typechecker.model.Element.getMatchingDeclarations(Element.java:244)
	at com.redhat.ceylon.model.typechecker.model.NamedArgumentList.getMatchingDeclarations(NamedArgumentList.java:38)
	at com.redhat.ceylon.ide.common.completion.completionManager_.getProposals$canonical$(IdeCompletionManager.ceylon:1733)
	at com.redhat.ceylon.ide.common.completion.completionManager_.getProposals(IdeCompletionManager.ceylon:1610)
	at com.redhat.ceylon.ide.common.completion.completionManager_.getContentProposals$canonical$(IdeCompletionManager.ceylon:269)
	at com.redhat.ceylon.ide.common.completion.completionManager_.getContentProposals(IdeCompletionManager.ceylon:114)
	at org.intellij.plugins.ceylon.ide.ceylonCode.completion.IdeaCompletionProvider.addCompletionsInternal$priv$(ideaCompletionManager.ceylon:211)
	at org.intellij.plugins.ceylon.ide.ceylonCode.completion.IdeaCompletionProvider.access$000(ideaCompletionManager.ceylon:68)
	at org.intellij.plugins.ceylon.ide.ceylonCode.completion.IdeaCompletionProvider$1.$call$(ideaCompletionManager.ceylon:173)
	at org.intellij.plugins.ceylon.ide.ceylonCode.model.concurrencyManager_.withIndexStrategy$priv$(concurrencyManager.ceylon:207)
	at org.intellij.plugins.ceylon.ide.ceylonCode.model.concurrencyManager_.withAlternateResolution(concurrencyManager.ceylon:219)
	at org.intellij.plugins.ceylon.ide.ceylonCode.completion.IdeaCompletionProvider.addCompletions(ideaCompletionManager.ceylon:140)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:36)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:155)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:81)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:132)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:125)
	at com.intellij.codeInsight.template.impl.LiveTemplateCompletionContributor$2.addCompletions(LiveTemplateCompletionContributor.java:90)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:36)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:155)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:81)
	at com.intellij.codeInsight.completion.CompletionService.performCompletion(CompletionService.java:110)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.calculateItems(CompletionProgressIndicator.java:776)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.access$500(CompletionProgressIndicator.java:90)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator$1CalculateItems.run(CompletionProgressIndicator.java:758)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$null$0(CompletionThreading.java:87)
	at com.intellij.codeInsight.completion.AsyncCompletion$$Lambda$895/1936912193.run(Unknown Source)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:848)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$null$1(CompletionThreading.java:84)
	at com.intellij.codeInsight.completion.AsyncCompletion$$Lambda$894/281172188.run(Unknown Source)
	at com.intellij.openapi.progress.impl.CoreProgressManager$3.run(CoreProgressManager.java:170)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:494)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:443)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:155)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$startThread$2(CompletionThreading.java:82)
	at com.intellij.codeInsight.completion.AsyncCompletion$$Lambda$893/450075408.run(Unknown Source)
	at com.intellij.openapi.application.impl.ApplicationImpl$2.run(ApplicationImpl.java:307)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by gavinking** [comment]
Apparently my fix to #460 didn't work. Which means that what we have here, *somehow* is a `Declaration` whose `container` is a `Package`, but whose `Scope` is null or has no name. So the underlying cause here is certainly a model loader bug. Who is forgetting to set `Scope`?

**Comment by karimiumar** [comment]
This is occurring every time the model loader/intellisense fails to recognize a keyword. If I'm working in a Java class, then the intellisense provides with a list of available options. But when I use a ceylon file then intellisense doesn't work as good as it works in Java and the above exception is reported. These keywords for which intellisense fails are not imported and as I expect the intellisense to provide with a list of packages to import the keyword from. For instance, if I'm typing
`preAuthorize` and if the package `org.springframework.security.access.prepost {
    preAuthorize
}` is not imported yet, then the exception occurs.

**Comment by gavinking** [comment]
@karimiumar I *believe* I have fixed it with my recent commits.

As you can see in #460, it's something we have also occasionally experienced, but we weren't able to reproduce it reliably in order to track down the cause.

I think it should be fixed now, but you would have to do a full build of Ceylon and of the IntelliJ plugin (i.e. `ant dist sdk intellij`) to test it.

**Comment by bjansen** [comment]
@gavinking I think you'll have to cherry pick those fixes in [this branch](https://github.com/ceylon/ceylon/tree/version-1.3.0), otherwise they won't be included if I release Ceylon IDE 1.3.0-1 (still based on Ceylon 1.3.0).

**Comment by gavinking** [comment]
@bjansen well the commits that fixed this aren't changes to IDE code. They're fixes to the `ceylon` project. Now, perhaps [this](https://github.com/ceylon/ceylon-ide-common/commit/e9917fecc9f6031c160506a047bc1629af8c748c) is a workaround for the problem, I'm not sure, I have not tested it as a fix to this issue. Anyway that is on common which has no branch.

**Comment by gavinking** [comment]
@bjansen It might be better to just catch the exception.

**Comment by bjansen** [comment]
> the commits that fixed this aren't changes to IDE code

That's why I pointed you to the ceylon/ceylon project ;)

**Comment by bjansen** [comment]
OK, I merged https://github.com/ceylon/ceylon-ide-common/commit/e9917fecc9f6031c160506a047bc1629af8c748c in [the 1.3.0-fixes branch](https://github.com/ceylon/ceylon-ide-common/tree/1.3.0-fixes), I think it's enough to fix this issue. Even if additional changes mades to `ceylon/ceylon` are not shipped in 1.3.0-1, we still have a correct workaround.
